code/game/objects/structures/shower.dm

```diff
@@ -114,26 +117,26 @@
 
 /obj/machinery/shower/Crossed(atom/movable/AM)
 	..()
-	if(on)
+	if(on && reagents.total_volume)
 		wash_atom(AM)
 
-/obj/machinery/shower/proc/wash_atom(atom/A)
-	A.wash(CLEAN_RAD | CLEAN_TYPE_WEAK) // Clean radiation non-instantly
-	A.wash(CLEAN_WASH)
-	SEND_SIGNAL(A, COMSIG_ADD_MOOD_EVENT, "shower", /datum/mood_event/nice_shower)
-	reagents.expose(A, TOUCH)
-	if(isliving(A))
-		check_heat(A)
+/obj/machinery/shower/proc/wash_atom(atom/target)
+	target.wash(CLEAN_RAD | CLEAN_TYPE_WEAK) // Clean radiation non-instantly
+	target.wash(CLEAN_WASH)
+	SEND_SIGNAL(target, COMSIG_ADD_MOOD_EVENT, "shower", /datum/mood_event/nice_shower)
+	reagents.expose(target, (TOUCH), SHOWER_EXPOSURE_MULTIPLIER * min(SHOWER_SPRAY_VOLUME / (reagents.total_volume || SHOWER_SPRAY_VOLUME), 1))
```

Comment:
this is some cryptic code
what does it do?

Reply:
the `a || b` operator, does it return boolean?

Reply:
The `||` operator basically just grabs the first thing that isn't falsey. This means that if the thing before it is `null` or `0` it will try to grab the thing after it. You can keep chaining them, but in this case it's just to prevent div/0 errors.

Reply:
it's one of the more obscure operators, is there another one you could use here?

Reply:
`max()` works, I'll use that.